feat(new-links): Parse url data and display

borismelnik opened the pull request:
Fixes: #10852

### What does the PR do

Unfurling new urls formats

### Affected areas

Invitations

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [ ] I've checked the design and this PR matches it


![Снимок экрана 2023-07-20 в 22 45 17](https://github.com/status-im/status-desktop/assets/82511785/69e5fd52-cf35-41e3-af61-a13c42758780)
![Снимок экрана 2023-07-24 в 21 51 56](https://github.com/status-im/status-desktop/assets/82511785/7cbb6e3f-42f2-4bfc-87e1-01d3c13ddda2)

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 56218022 | [#1](https://ci.status.im/job/status-desktop/job/branches/job/linux/job/x86_64/job/tests-nim/job/PR-11619/1/) | 2023-07-20 19:55:34 | ~5 min | `tests/nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/branches/job/linux/job/x86_64/job/tests-nim/job/PR-11619/1//consoleText) |

jrainville commented on src/app/modules/main/module.nim:
```diff
@@ -1256,3 +1258,31 @@ method windowActivated*[T](self: Module[T]) =
+  jsonObj["displayName"] = %* contactData.displayName
+  jsonObj["description"] = %* contactData.description
+  jsonObj["publicKey"] = %* contactData.publicKey
+  return $jsonObj
```
Can we put those in the `urls_manager` instead?
I think we just need to expose it to the QML using `setRootContextProperty`.
Though I think it can't have access to services.

Anyway, maybe someone has an idea where we could put them? Putting them in the main module works, it just feels like it's a bit random

MishkaRogachev replied:
Having a url service and mvc makes sense

borismelnik replied:
so probably, I should move generators to new url service as well? otherwise, its service with just one method. @jrainville @MishkaRogachev what do you think?

MishkaRogachev replied:
Yeah, I do believe that urls could become massive in the future

jrainville reviewed (commented):
Looks good.

I added one comment about the functions probably not belonging in the main module, but I don't really have a better option form the top of my head.

I had two questions though:
1. does the "Start chat" work with the new urls too? Like pasting the profile url of someone should open their chat or the contact request popup 
2. Does the community data in the unfurl update if the community info changes? For example, if the community name changes after it unfurled, does it change to use the new name?

borismelnik commented:
> Looks good.
> 
> I added one comment about the functions probably not belonging in the main module, but I don't really have a better option form the top of my head.
> 
> I had two questions though:
> 
> 1. does the "Start chat" work with the new urls too? Like pasting the profile url of someone should open their chat or the contact request popup
> 2. Does the community data in the unfurl update if the community info changes? For example, if the community name changes after it unfurled, does it change to use the new name?



> Looks good.
> 
> I added one comment about the functions probably not belonging in the main module, but I don't really have a better option form the top of my head.
> 
> I had two questions though:
> 
> 1. does the "Start chat" work with the new urls too? Like pasting the profile url of someone should open their chat or the contact request popup
> 2. Does the community data in the unfurl update if the community info changes? For example, if the community name changes after it unfurled, does it change to use the new name?

unfortunately, i forgot about start chat... will do.

jrainville commented on src/app/modules/main/controller.nim:
```diff
@@ -22,6 +22,7 @@ import ../../../app_service/service/wallet_account/service as wallet_account_ser
 import ../../../app_service/service/token/service as token_service
 import ../../../app_service/service/network/service as networks_service
 import ../../../app_service/service/collectible/service as collectible_service
+import ../../../app_service/service/general/service as general_service
```
I think this one is no longer needed

jrainville commented on ui/imports/utils/Utils.qml:
```diff
@@ -515,6 +516,16 @@ QtObject {
+        }
+
+        let communityDataString = sharedUrlsModuleInst.parseCommunitySharedUrl(link)
+        return JSON.parse(communityDataString)
```
Might need a `try/catch`?

jrainville reviewed (approved):
Nice job

igor-sirotin commented:
Currently community link previews are not shown in chat. I commented them out to implement after this:
- https://github.com/status-im/status-desktop/issues/11634

@jrainville do you think we can merge this without working link previews and I will take care of them when backend is ready?

jrainville commented:
> Currently community link previews are not shown in chat. I commented them out to implement after this:
> 
> * [Support new link previews for image and status links #11634](https://github.com/status-im/status-desktop/issues/11634)
> 
> @jrainville do you think we can merge this without working link previews and I will take care of them when backend is ready?

Sure, as long as clicking the link takes you to the community

MishkaRogachev reviewed (approved):
Great new service and mvc for shared links

caybro commented on src/app_service/service/general/service.nim:
```diff
@@ -100,4 +100,5 @@ QtObject:
-      error "error: ", procName="backupData", errName = e.name, errDesription = e.msg
\ No newline at end of file
+      error "error: ", procName="backupData", errName = e.name, errDesription = e.msg
+    
```
Stray empty line?

caybro replied:
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline ;)

caybro replied:
And btw, QtC can help here :)

![image](https://github.com/status-im/status-desktop/assets/5377645/3bec21ed-c81c-472d-8a11-d3bf03ae0a12)

caybro replied:
Pretty sure VSCode has something like that too

caybro commented on ui/app/AppLayouts/Chat/stores/RootStore.qml:
```diff
@@ -455,26 +455,28 @@ QtObject {
+        const communityName = getSectionNameById(result.communityData.communityId)
         if (!communityName) {
             // Unknown community, fetch the info if possible
             root.requestCommunityInfo(result.communityId)
```
@borismelnik actually this must be fixed

caybro commented on src/app/modules/main/controller.nim:
```diff
@@ -509,4 +509,4 @@ proc slowdownArchivesImport*(self:Controller) =
 
 proc speedupArchivesImport*(self:Controller) =
-  communityService.speedupArchivesImport()
+  communityService.speedupArchivesImport() 
```
Just revert the changes and don't touch the file :D

caybro reviewed (approved):
LGTM!

lukaszso reviewed (approved):
Tested and it works fine. As mentioned before the preview is not available but clicking on the link takes the user to the community. User links also work :)
